applications/asset_tracker_v2/src/modules/Kconfig.data_module

```diff
@@ -11,10 +11,6 @@ menuconfig DATA_MODULE
 
 if DATA_MODULE
 
-config DATA_AGPS_REQUEST_ALL_UPON_CONNECTION
```

Comment:
After doing some testing with ATv2 I realized that this feature is actually did make a difference and we probably should have something similar also with the Location library.

When ATv2 starts, data sampling starts immediately. This means that Location module asks Location library to get a position using GNSS. GNSS in turn requests A-GPS data and the A-GPS data request is tried to be sent to the cloud. Now, if the modem can't immediately connect to the network or the cloud connection can't be established quickly enough, requesting A-GPS data fails. This means that GNSS will do a cold start and depending on various things, GNSS may not be even able to get a fix. During my test run, GNSS failed to get a fix until GNSS requested A-GPS data again, 1 hour later.

Postponing data sampling was my first thought, but after discussing with @simensrostad, we probably want to keep the sampling as it is and buffer data until it can be sent to the cloud. In that case we should have some kind of solution to request A-GPS data when the cloud connection has been established.

Reply:
Based on what we have discussed today, we have two alternatives:

1. Use GNSS priority mode in location library. We'll try this out tomorrow. This should enable downloading of A-GPS data from the satellites more reliably when there are issues in LTE or cloud connection. And this is a useful feature anyway.
2. Add logic to ATv2 to send AGPS request to cloud only when cloud connection is available. This part is handled fine. However, if cloud connection is missing, we should buffer the request and send when the connection is established.

Reply:
I've created NCSDK-17197 on this so let's handle this topic outside of this PR.